lib/matplotlib/backend_tools.py

```diff
@@ -99,20 +98,6 @@ def trigger(self, sender, event, data=None):
 
         pass
 
-    @figure.setter
-    def figure(self, figure):
-        """
-        Set the figure
-
-        Set the figure to be affected by this tool
-
-        Parameters
-        ----------
-        figure: `Figure`
-        """
-
-        self._figure = figure
```

Comment:
I still don't feel convinced about this, I see ``FigureManager``, while a lot more flexible then before, it still functions very much as a neat and easy wrapper, as soon as you try to do anything complicated, like embedding, it falls apart and thus we shouldn't build MEP22 around it.

I would much rather emit an event from ``ToolManager`` when the figure changes and/or don't let the tool keep the figure... ooh, actually, scrap that last bit, we can build in a figure updater callback into ``ToolBase`` and descendent classes can override that updater method if they need to do anything extra.

If we go with that then I don't think we need to implement that now as we purely add a new feature, we don't change any existing structure, so it will remain fully BC.